### PR TITLE
Basic structure

### PR DESCRIPTION
**Pull request by terminatorbill**

### CODE REVIEW

**Comment by terminatorbill** [comment]
@chriniko13 Please check the pit coverage and why it is 0%. I suspect something weird it's going on and maybe the cause is Junit5 regardless of the dependencies I added so that other frameworks may use Junit 4.12 when they do not support JUnit5 yet.

**Comment by terminatorbill** [comment]
@chriniko13 Also evaluate whether we need the integration test and if yes then what should we test there.

File: pom.xml [hunk 1]
```diff
@@ -12,7 +12,7 @@
     <description>Sample project which shows the absolutely minimum for good engineering</description>
 
     <properties>
-        <jacoco.version>0.7.7.201606060606</jacoco.version>
+        <jacoco.version>0.8.1</jacoco.version>
```
Comment (chriniko13) [inline pom.xml:15]: :1st_place_medal: @terminatorbill you rock!!!

File: pom.xml [hunk 2]
```diff
@@ -173,7 +202,7 @@
             <plugin>
                 <groupId>org.pitest</groupId>
                 <artifactId>pitest-maven</artifactId>
-                <version>1.1.10</version>
+                <version>1.4.2</version>
```
Comment (chriniko13) [inline pom.xml:205]: :1st_place_medal: @terminatorbill again you rock!!!

File: src/main/java/com/brogrammers/Application.java [hunk 3]
```diff
@@ -1,13 +1,33 @@
...
+import com.brogrammers.domain.BrogrammerRegistry;
+import com.brogrammers.domain.BrogrammerValidNameFilter;
 
 public class Application {
 
     public static void main(String[] args) {
 
-        Brogrammer
+        BrogrammerFilter brogrammerFilter = new BrogrammerValidNameFilter(Pattern.compile("[a-zA-Z]{3,25}"));
+        Properties sample = loadProperties();
```
Comment (chriniko13) [inline src/main/java/com/brogrammers/Application.java:15]: :+1: nice extraction for properties file!

File: src/test/java/com/brogrammers/domain/BrogrammerValidNameFilterTest.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        // when
+        boolean shouldKeep = brogrammerFilter.shouldKeep(brogrammer);
+
+        // then
+        assertThat(shouldKeep, is(isValid));
+    }
+
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> createBrogrammerNames() {
```
Comment (chriniko13) [inline src/test/java/com/brogrammers/domain/BrogrammerValidNameFilterTest.java:41]: :+1: you run a lot of extra miles!!!

**Review by chriniko13 — APPROVED** [review]
You rock!!!

**Pull request merged**